keydata_test.go

```diff
@@ -244,28 +248,35 @@ func (s *keyDataTestBase) TearDownSuite(c *C) {
 	RegisterPlatformKeyDataHandler(mockPlatformName, nil)
 }
 
-func (s *keyDataTestBase) newKeyDataKeys(c *C, sz1, sz2 int) (DiskUnlockKey, AuxiliaryKey) {
-	key := make([]byte, sz1)
-	auxKey := make([]byte, sz2)
-	_, err := rand.Read(key)
+func (s *keyDataTestBase) newPrimaryKey(c *C, sz1 int) PrimaryKey {
+	primaryKey := make([]byte, sz1)
+	_, err := rand.Read(primaryKey)
 	c.Assert(err, IsNil)
-	_, err = rand.Read(auxKey)
-	c.Assert(err, IsNil)
-	return key, auxKey
+
+	return PrimaryKey(primaryKey)
 }
 
-func (s *keyDataTestBase) mockProtectKeys(c *C, key DiskUnlockKey, auxKey AuxiliaryKey, modelAuthHash crypto.Hash) (out *KeyParams) {
-	payload := MarshalKeys(key, auxKey)
+func (s *keyDataTestBase) mockProtectKeys(c *C, primaryKey PrimaryKey, modelAuthHash crypto.Hash) (out *KeyParams, unlockKey DiskUnlockKey) {
+	unique := make([]byte, len(primaryKey))
+	_, err := rand.Read(unique)
+	c.Assert(err, IsNil)
+
+	reader := new(bytes.Buffer)
+	reader.Write(unique)
+
+	unlockKey, payload, err := MakeDiskUnlockKey(reader, crypto.SHA256, primaryKey)
```

Comment:
You should just be able to pass `bytes.NewReader(unique)` as the first argument here.

Reply:
Thanks, fixed in 2 places (+1 in a later PR).